src/doc/unstable-book/src/compiler-flags/function-return.md

```diff
@@ -0,0 +1,25 @@
+# `function-return`
+
+The tracking issue for this feature is: https://github.com/rust-lang/rust/issues/116853.
+
+------------------------
+
+Option `-Zfunction-return` controls how function returns are converted.
+
+It is equivalent to [Clang]'s and [GCC]'s `-mfunction-return`. The Linux kernel
+uses it for RETHUNK builds. For details, see [LLVM commit 2240d72f15f3] ("[X86]
+initial -mfunction-return=thunk-extern support") which introduces the feature.
+
+Supported values for this option are:
+
+  - `keep`: do not convert function returns.
+  - `thunk-extern`: convert function returns (`ret`) to jumps (`jmp`)
+    to an external symbol called `__x86_return_thunk`.
+
+Like in Clang, GCC's values `thunk` and `thunk-inline` are not supported.
+
+Only x86 is supported.
```

Comment:
Maybe add an error somewhere in rustc when trying to use it outside of x86?

Reply:
Yeah, it would be probably be best to do so to avoid mistakes -- see https://github.com/rust-lang/rust/pull/116892#issuecomment-1768729744. GCC and Clang both also check that one is not using the `large` code model, so I will also add that.

Reply:
I was talking to @nbdd0121 and he told me that a simple check is probably enough, i.e. no need to add state to the target spec if all x86 targets are supposed to support it.

Reply:
Done, please take a look.